orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -394,36 +407,65 @@ def _get_exported_model(self, input_schema, *inputs, **kwargs):
+
+                   required_export_kwargs = {'input_names': self._input_info.names,
+                                              'output_names': output_names,
+                                              'opset_version': ortmodule.ONNX_OPSET_VERSION,
+                                              'do_constant_folding': False,
+                                              'training': self._export_mode,
+                                              'dynamic_axes': self._input_info.dynamic_axes,
+                                              'verbose': self._debug_options.logging.log_level < LogLevel.WARNING,
+                                              'export_params': False,
+                                              'keep_initializers_as_inputs': True}
+                   invalid_args = self._export_extra_kwargs.keys() & required_export_kwargs.keys()
+                   assert len(invalid_args) == 0,\
+                       f"The following PyTorch exporter arguments cannot be specified: '{invalid_args}'."
+                   torch.onnx.export(self._flattened_module,
+                                     sample_inputs_as_tuple,
+                                     f,
+                                     **required_export_kwargs,
+                                     **self._export_extra_kwargs)
+           except Exception as e:
+               raise wrap_exception(ORTModuleONNXModelException,
+                                    RuntimeError(f'There was an error while exporting the PyTorch model to ONNX: '
+                                                 f'\n\n{_utils.get_exception_as_string(e)}'))
+           exported_model = onnx.load_model_from_string(f.getvalue())
+
+           exported_model = _post_process_after_export(exported_model,
                                                     self._enable_custom_autograd_function,
                                                     self._debug_options.logging.log_level)
+        else:
+           try:
+               with torch.no_grad():
+                   required_export_kwargs = {'input_names': self._input_info.names,
+                                             'output_names': output_names,
+                                             'opset_version': ortmodule.ONNX_OPSET_VERSION,
+                                             'do_constant_folding': True,
+                                             'training': self._export_mode,
+                                             'dynamic_axes': self._input_info.dynamic_axes,
+                                             'verbose': self._debug_options.logging.log_level < LogLevel.WARNING,
+                                             'operator_export_type': OperatorExportTypes.ONNX_ATEN_FALLBACK,
+                                             'export_params': True,
+                                             'keep_initializers_as_inputs': False}
```

Comment:
Doing this implies that a lot of the `torch.nn.Module` functionality is no longer usable. For example, users cannot query `model.named_parameters()` since they will be different from the initializers being used (inside the exported onnx model).
This might be ok for the inference scenario (because the parameters are not going to change in inference). But the branching in this code here is happening on the provider, not on the training mode.